fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java

```diff
@@ -2960,4 +2958,54 @@ public long loadDb(DataInputStream dis, long checksum) throws IOException, DdlEx
     public ConcurrentHashMap<Long, Database> getIdToDb() {
         return new ConcurrentHashMap<>(idToDb);
     }
+
+    private void createPartitionWithIndicesElasticOnStorageMedium(String clusterName, long dbId, long tableId,
+            long baseIndexId, long partitionId, String partitionName, Map<Long, MaterializedIndexMeta> indexIdToMeta,
+            DistributionInfo distributionInfo, TStorageMedium storageMedium, ReplicaAllocation replicaAlloc,
+            Long versionInfo, Set<String> bfColumns, double bfFpp, Set<Long> tabletIdSet, List<Index> indexes,
+            boolean isInMemory, TStorageFormat storageFormat, TTabletType tabletType, TCompressionType compressionType,
+            DataSortInfo dataSortInfo, boolean enableUniqueKeyMergeOnWrite, String storagePolicy,
+            IdGeneratorBuffer idGeneratorBuffer, boolean disableAutoCompaction, boolean enableSingleReplicaCompaction,
+            boolean skipWriteIndexOnLoad, boolean storeRowColumn, boolean isDynamicSchema, BinlogConfig binlogConfig,
+            OlapTable olapTable, PartitionInfo partitionInfo, long bufferSize) throws DdlException {
+
+        try {
+            // create partition
+            Partition partition = createPartitionWithIndices(clusterName, dbId, tableId,
+                    baseIndexId, partitionId, partitionName, indexIdToMeta, distributionInfo, storageMedium,
+                    replicaAlloc, versionInfo, bfColumns, bfFpp, tabletIdSet, indexes, isInMemory, storageFormat,
+                    tabletType, compressionType, dataSortInfo, enableUniqueKeyMergeOnWrite, storagePolicy,
+                    idGeneratorBuffer, disableAutoCompaction, enableSingleReplicaCompaction, skipWriteIndexOnLoad,
+                    storeRowColumn, isDynamicSchema, binlogConfig);
+            olapTable.addPartition(partition);
+        } catch (DdlException e) {
+            // check whether set storage medium from data property
```

Comment:
add comment here to tell we are using different medium because shortage on the original medium